base/compiler/tfuncs.jl

```diff
@@ -1086,22 +1102,38 @@ function apply_type_nothrow(argtypes::Array{Any, 1}, @nospecialize(rt))
     for i = 2:length(argtypes)
         isa(u, UnionAll) || return false
         ai = widenconditional(argtypes[i])
-        if ai ⊑ TypeVar
+        if ai ⊑ TypeVar || ai === DataType
```

Comment:
Probably not true since `Vararg` isa DataType.

Reply:
:(

Reply:
@JeffBezanson and I talked about fixing this by making `Vararg` not a DataType, so we'll wait with this PR until @JeffBezanson gets to that change.